modules/services/window-managers/hyprland.nix

```
@@ -0,0 +1,113 @@
+{ config, lib, pkgs, ... }:
+let cfg = config.wayland.windowManager.hyprland;
+in {
+  meta.maintainers = [ lib.maintainers.fufexan ];
+
+  options.wayland.windowManager.hyprland = {
+    enable = lib.mkEnableOption (lib.mdDoc "Hyprland wayland compositor");
+
+    package = lib.mkPackageOptionMD pkgs "hyprland" { };
+
+    finalPackage = cfg.package.override {
+      enableXWayland = cfg.xwayland.enable;
+      hidpiXWayland = cfg.xwayland.hidpi;
+      nvidiaPatches = cfg.enableNvidiaPatches;
+    };
```

Unfortunately, it is a bit more complicated. Need something like

```suggestion
    finalPackage = lib.mkOption {
      type = lib.types.package;
      readOnly = true;
      default = cfg.package.override {
        enableXWayland = cfg.xwayland.enable;
        hidpiXWayland = cfg.xwayland.hidpi;
        nvidiaPatches = cfg.enableNvidiaPatches;
      };
      defaultText = lib.literalMD
        "`wayland.windowManager.hyprland.package` with applied configuration";
      description = ''
        The Hyprland package after applying configuration.
      '';
    };
```

---

Oops, forgot about that.